### PR TITLE
Add --raw-output-data-prefix to the pyflyte-execute command

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Please see the linked issue for more information.  This adds a `--raw-output-data-prefix` switch to the pyflyte-execute command and container args will be changed so that they're registered with `--raw-data-output-prefix {{.rawOutputDataPrefix}}`.  Propeller will then fill them in.

*Release this change last, after testing propeller/plugins*

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
* The AWS and GCP data proxy classes now take in its constructor a prefix which will be used when `get_random_path` is called.  If it is missing it will fall back to the current behavior of looking up either the S3 shard formatter config or the GCS prefix config.
* Added switch to click command
* Added switch and template placeholder to sdk runnable task and notebook task container args.

## Tracking Issue
https://github.com/lyft/flyte/issues/211

## Follow-up issue
NA

### CODE REVIEW

File: flytekit/common/tasks/sdk_runnable.py [hunk 1]
```diff
@@ -447,6 +447,8 @@ def _get_container_definition(
                 "{{.input}}",
                 "--output-prefix",
                 "{{.outputPrefix}}",
+                "--raw-data-output-prefix",
+                "{{.outputPrefix}}"
```
Comment (EngHabu) [inline flytekit/common/tasks/sdk_runnable.py:451]: This should be raw output prefix, right?

File: flytekit/bin/entrypoint.py [hunk 2]
```diff
@@ -97,10 +100,11 @@ def _pass_through():
 @_click.option("--task-name", required=True)
 @_click.option("--inputs", required=True)
 @_click.option("--output-prefix", required=True)
+@_click.option("--raw-data-output-prefix", required=False)
 @_click.option("--test", is_flag=True)
-def execute_task_cmd(task_module, task_name, inputs, output_prefix, test):
+def execute_task_cmd(task_module, task_name, inputs, output_prefix, raw_output_data_prefix, test):
     _click.echo(_utils.get_version_message())
-    _execute_task(task_module, task_name, inputs, output_prefix, test)
+    _execute_task(task_module, task_name, inputs, output_prefix, raw_output_data_prefix, test)
```
Comment (EngHabu) [inline flytekit/bin/entrypoint.py:107]: Discussed offline: check for the hardcoded "$RawDataOutputPrefix" and if so, set it to None.

File: flytekit/common/tasks/sdk_runnable.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/common/tasks/sdk_runnable.py:451]: yeah this is when i realized that flyteplugins wasn't done

**Comment by kumare3** [comment]
Hurrah! One step closer to portable Workflows. Definitely all the examples will be portable

File: flytekit/bin/entrypoint.py [hunk 3]
```diff
@@ -83,7 +83,8 @@ def _execute_task(task_module, task_name, inputs, output_prefix, test):
                 _data_proxy.Data.get_data(inputs, local_inputs_file)
                 input_proto = _utils.load_proto_from_file(_literals_pb2.LiteralMap, local_inputs_file)
                 _engine_loader.get_engine().get_task(task_def).execute(
-                    _literal_models.LiteralMap.from_flyte_idl(input_proto), context={"output_prefix": output_prefix},
+                    _literal_models.LiteralMap.from_flyte_idl(input_proto),
```
Comment (kumare3) [inline flytekit/bin/entrypoint.py:86]: wow this seems some hairy logic
